<!DOCTYPE html>
<html>
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/gitalk@1/dist/gitalk.css">
<script src="https://cdn.jsdelivr.net/npm/gitalk@1/dist/gitalk.min.js"></script>
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>A universal SNP and small-indel variant caller using deep neural networks - DawsonWen的个人网站</title>
    <meta name="author"  content="DawsonWen">
    <meta name="description" content="A universal SNP and small-indel variant caller using deep neural networks">
    <meta name="keywords"  content="论文阅读">
    <!-- Open Graph -->
    <meta property="og:title" content="A universal SNP and small-indel variant caller using deep neural networks - DawsonWen的个人网站">
    <meta property="og:type" content="website">
    <meta property="og:url" content="http://localhost:4000/2024/08/29/deepvariant.html">
    <meta property="og:description" content="为天地立心, 为生民立命, 为往圣继绝学, 为万世开太平">
    <meta property="og:site_name" content="DawsonWen的个人网站">
    <link rel="stylesheet" href="//cdn.staticfile.org/normalize/6.0.0/normalize.min.css">
    <link rel="stylesheet" href="//at.alicdn.com/t/font_roc50gemkxpw4s4i.css">
    <link rel="stylesheet" href="/assets/css/github-markdown.css">
    <link rel="stylesheet" href="/assets/css/prism.css">
    <link rel="stylesheet" href="/assets/css/share.min.css">
    <link rel="stylesheet" href="/assets/css/app.min.css">
    <link rel="stylesheet" href="https://cdn.bootcss.com/font-awesome/4.7.0/css/font-awesome.min.css">
    <script src="https://cdn.staticfile.org/jquery/3.2.1/jquery.min.js"></script>
	
	<!--
Author: Ray-Eldath
refer to:
 - http://docs.mathjax.org/en/latest/options/index.html
-->

	<script type="text/javascript" async src="https://cdn.bootcss.com/mathjax/2.7.2/MathJax.js?config=TeX-MML-AM_CHTML"></script>
	
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({
		jax: ["input/TeX", "output/HTML-CSS"],
		tex2jax: {
			inlineMath: [ ["$", "$"], ["\\(","\\)"] ],
			displayMath: [ ["$$", "$$"], ["\\[","\\]"] ],
			skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
		},
		"HTML-CSS": { preferredFont: "TeX", availableFonts: ["STIX","TeX"] }
      });
    </script>


	
    <!--
Author: Ray-Eldath
-->
<style>
    .markdown-body .anchor{
        float: left;
        margin-top: -8px;
        margin-left: -20px;
        padding-right: 4px;
        line-height: 1;
        opacity: 0;
    }
    
    .markdown-body .anchor .anchor-icon{
        font-size: 15px
    }
</style>
<script>
    $(document).ready(function() {
        let nodes = document.querySelector(".markdown-body").querySelectorAll("h1,h2,h3")
        for(let node of nodes) {
            var anchor = document.createElement("a")
            var anchorIcon = document.createElement("i")
            anchorIcon.setAttribute("class", "fa fa-anchor fa-lg anchor-icon")
            anchorIcon.setAttribute("aria-hidden", true)
            anchor.setAttribute("class", "anchor")
            anchor.setAttribute("href", "#" + node.getAttribute("id"))
            
            anchor.onmouseover = function() {
                this.style.opacity = "0.4"
            }
            
            anchor.onmouseout = function() {
                this.style.opacity = "0"
            }
            
            anchor.appendChild(anchorIcon)
            node.appendChild(anchor)
        }
    })
</script>
	
    <script>
        var _hmt = _hmt || [];
        (function() {
            var hm = document.createElement("script");
            hm.src = "https://hm.baidu.com/hm.js?671e6ffb306c963dfa227c8335045b4f";
            var s = document.getElementsByTagName("script")[0]; 
            s.parentNode.insertBefore(hm, s);
		
        })();
    </script>

</head>


<body>
  <!--[if lt IE 10]>
<div class="alert-danger" role="alert">你的浏览器实在太太太旧了，放学别走，升级完浏览器再说！<a target="_blank" class="alert-link" href="http://browsehappy.com">立即升级</a></div>
<![endif]-->
  <input id="nm-switch" type="hidden" value="true"> <header class="g-header">
    <div class="g-logo">
      <a href="/"></a>
    </div>
    <i id="menu-toggle" class="iconfont icon-menu"></i>
    <nav class="g-nav">
        <ul>
            
            <li><a href="/">home</a></li>
            
            <li><a href="/tags.html">tags</a></li>
            
        </ul>
    </nav>
</header>


  <header
    class="g-banner post-header post-pattern-circuitBoard bgcolor-default "
    data-theme="default"
  >
    <div class="post-wrapper">
      <div class="post-tags">
        
          
            <a href="/tags.html#%E8%AE%BA%E6%96%87%E9%98%85%E8%AF%BB" class="post-tag">论文阅读</a>
          
        
      </div>
      <h1>A universal SNP and small-indel variant caller using deep neural networks</h1>
      <div class="post-meta">
        <span class="post-meta-item"><i class="iconfont icon-author"></i>郑之杰</span>
        <time class="post-meta-item" datetime="24-08-29"><i class="iconfont icon-date"></i>29 Aug 2024</time>
      </div>
    </div>
    
    <div class="filter"></div>
      <div class="post-cover" style="background: url('https://pic.imgdb.cn/item/66d0605fd9c307b7e91b1e7b.png') center no-repeat; background-size: cover;"></div>
    
  </header>

  <div class="post-content visible">
    

    <article class="markdown-body">
      <blockquote>
  <p>使用深度神经网络的通用单核苷酸多态性和插入/缺失变异比对器.</p>
</blockquote>

<ul>
  <li>paper：<a href="https://www.nature.com/articles/nbt.4235">A universal SNP and small-indel variant caller using deep neural networks</a></li>
  <li>code：<a href="https://github.com/google/deepvariant">DeepVariant</a></li>
</ul>

<h1 id="1-背景介绍">1. 背景介绍</h1>

<p>单核苷酸多态性（<strong>Single Nucleotide Polymorphism，SNP</strong>）是指基因组中单个核苷酸腺嘌呤(<strong>A</strong>)、胸腺嘧啶(<strong>T</strong>)、胞嘧啶(<strong>C</strong>)或鸟嘌呤(<strong>G</strong>)在个体之间或个体配对染色体之间的差异，是造成基因组多样性的最常见的<strong>DNA</strong>序列变异。</p>

<p>插入缺失（<strong>insertion-deletion，InDel</strong>）是指在<strong>DNA</strong>序列中添加或删除少量碱基，主要指在基因组某个位置上发生较短长度的线性片段插入(<strong>Insert</strong>)或者缺失(<strong>Deletion</strong>)的现象，也是一种常见的<strong>DNA</strong>序列变异。</p>

<p><img src="https://pic.imgdb.cn/item/66d03dffd9c307b7e9efeaa1.png" alt="" /></p>

<p><strong>SNP</strong>和<strong>INDEL</strong>变异检测有助于我们更深入地了解基因组，生物性状的表现，物种的起源与进化，认识基因变异和疾病的之间的联系。从测序数据中进行准确的变异检测也是生物学、医学研究和精准医学的基础。</p>

<h1 id="2-deepvariant">2. DeepVariant</h1>

<p><strong>DeepVariant</strong>是由<strong>Google Brain</strong>开发的一个开源软件，用于对基因组序列进行变异检测。它利用深度学习技术来实现高质量、高准确性的变异检测，能够在人类基因组的全长上实现高达99％的准确率。</p>

<p><strong>DeepVariant</strong>使用卷积神经网络（<strong>CNN</strong>）来对原始<strong>DNA</strong>测序数据进行分析，并识别基因组中的单核苷酸变异（<strong>SNP</strong>）和插入/缺失（<strong>indel</strong>）。<strong>DeepVariant</strong>的基本工作流程有三个步骤：</p>
<ol>
  <li><strong>make_examples</strong>: 将测序数据（经过比对的<strong>bam</strong>文件）中可能存在突变的候选位点编码成<strong>pileup image</strong>。</li>
  <li><strong>call_variants</strong>: 调用预训练的<strong>Google Inception V2</strong>模型，输出候选位点对应的三种不同基因型的概率。</li>
  <li><strong>postprocess_variants</strong>: 对候选位点进行合并处理，输出最终的突变检测<strong>VCF</strong>文件。</li>
</ol>

<p><img src="https://pic.imgdb.cn/item/66d03f3dd9c307b7e9f0fa2d.png" alt="" /></p>

<p><strong>pileup image</strong>是根据<strong>reads</strong>提供的突变个数生成的图像，候选点位始终位于图像的中心。比如候选位点参考基因组碱基为<strong>A</strong>，测序得到的<strong>reads</strong>显示该位点含有<strong>G</strong>和<strong>T</strong>两种，因此可能存在3种等位基因(<strong>{G,T,G&amp;T}</strong>)，因此生成3张图像。并结合真实的突变信息进行标注，一共有3种标签：</p>
<ul>
  <li><strong>0/homozygous reference</strong>：纯合子并且和参考基因组相同；</li>
  <li><strong>1/heterozygous</strong>：杂合子；</li>
  <li><strong>2/homozygous alternative</strong>：纯合子但与参考基因组不同。</li>
</ul>

<p>由于全基因组测序数据的平均覆盖深度一般不会超过$100$，因此<strong>pileup image</strong>图像高度设置为$100$；对于测序覆盖度大于$100$的区域，则通过下采样技术将其覆盖度降低至$100$以满足图像生成的要求。此时最多可编码$95$个<strong>read</strong>，顶部的$5$行保留用于参考序列。</p>

<p>现有主流测序平台产生的测序数据读长在$100bp$左右，对于候选的变异位点，向基因组两端延伸$100bp$即可将大部分覆盖待检位点的测序数据包含进来；考虑到插入缺失变异的长度，为其预留$20bp$的编码空间，因此<strong>pileup image</strong>图像宽度设置为$221$。</p>

<p><strong>pileup image</strong>图像编码为$6$个通道：</p>
<ol>
  <li><strong>read base</strong>：碱基类型，不同强度代表<strong>A,C,G,T</strong></li>
  <li><strong>base quality</strong>：碱基的测序质量，由测序仪设定，数值越大表示质量较高</li>
  <li><strong>mapping quality</strong>：<strong>read</strong>的比对质量，由对齐器设置，数值越大表示质量较高</li>
  <li><strong>strand</strong>：<strong>read</strong>链的方向；灰色为正向，白色为反向</li>
  <li><strong>read supports variant</strong>：<strong>read</strong>是否支持变异；白色表示支持给定的替代等位基因，灰色表示不支持</li>
  <li><strong>base differ from ref</strong>：<strong>read</strong>链与<strong>ref</strong>链对应位置碱基是否相同；白色表示与参考值不同，灰色表示与参考值匹配</li>
</ol>

<p><img src="https://pic.imgdb.cn/item/66d05b05d9c307b7e9165d8f.png" alt="" /></p>

<p><strong>DeepVariant</strong>采用的<strong>CNN</strong>网络的输入层尺寸是$299×299$，因此需要对原始图像($221×100×6$)的空间尺寸进行缩放。</p>

    </article>

    
    <div class="social-share-wrapper">
      <div class="social-share"></div>
    </div>
    
  </div>

  <section class="author-detail">
    <section class="post-footer-item author-card">
      <div class="avatar">
        <img src="https://avatars.githubusercontent.com/u/46283762?v=4&size=64" alt="">
      </div>
      <div class="author-name" rel="author">DawsonWen</div>
      <div class="bio">
        <p></p>
      </div>
      
      <ul class="sns-links">
        
        <li>
          <a href="//github.com/Sologala" target="_blank">
                    <i class="iconfont icon-github"></i>
                </a>
        </li>
        
      </ul>
      
    </section>
    <section class="post-footer-item read-next">
      
      <div class="read-next-item">
        <a href="/2024/08/30/deBruijn.html" class="read-next-link"></a>
        <section>
          <span>欧拉路径(Euler Path)与de Bruijn图</span>
          <p>  Euler Path and de Bruijn Graph.</p>
        </section>
        
        <div class="filter"></div>
        <img src="https://pic.imgdb.cn/item/66d1a018d9c307b7e9ae1653.png" alt="">
        
     </div>
      

      
      <div class="read-next-item">
        <a href="/2024/07/16/loraga.html" class="read-next-link"></a>
          <section>
            <span>LoRA-GA: Low-Rank Adaptation with Gradient Approximation</span>
            <p>  LoRA-GA：梯度近似的低秩参数高效微调.</p>
          </section>
          
          <div class="filter"></div>
          <img src="https://pic.imgdb.cn/item/669621f1d9c307b7e9b33d48.png" alt="">
          
      </div>
      
    </section>
    
    <section class="post-footer-item comment">
      <div id="disqus_thread"></div>
      <div id="gitalk_container"></div>
    </section>
  </section>

  <!-- <footer class="g-footer">
  <script type='text/javascript' id='clustrmaps' src='//cdn.clustrmaps.com/map_v2.js?cl=ffffff&w=800&t=m&d=WWuzUTmOt8V9vdtIQd5uqrEcKsRg4IiPuy9gg21CQO8'></script>
  <section>DawsonWen的个人网站 ©
  
  
    2020
    -
  
  2024
  </section>
  <section>Powered by <a href="//jekyllrb.com">Jekyll</a></section>
</footer>
 -->

  <script src="/assets/js/social-share.min.js"></script>
  <script src="https://cdn.jsdelivr.net/npm/gitalk@1/dist/gitalk.min.js"></script>
  <script>
    socialShare('.social-share', {
      sites: [
        
          'wechat'
          ,
          
        
          'weibo'
          ,
          
        
          'douban'
          ,
          
        
          'twitter'
          
        
      ],
      wechatQrcodeTitle: "分享到微信朋友圈",
      wechatQrcodeHelper: '<p>扫码后点击右上角</p><p>将本文分享至朋友圈</p>'
    });
  </script>

  
	
  

  <script src="/assets/js/prism.js"></script>
  <script src="/assets/js/index.min.js"></script>
</body>

</html>
